Algo.h

```diff
@@ -294,7 +385,8 @@ class MMP
                 // invert wale pains???
 
                 for (auto &e : ((Vertex *)(source->p))->edges) {
-                    Interval ii(0, 0, 0, e->length(), 0, NULL, e);
+                    //check for invert
+                    Interval ii(0, 0, 0, e->length(), 0, NULL, e, false);
```

Comment:
We **are** creating new intervals. Baad mein you won't know when this interval was constructed (the context) and whether or not it should be inverted.